dev-tools/omdb/src/bin/omdb/db.rs

```diff
@@ -1426,6 +1528,431 @@ async fn cmd_db_snapshot_info(
+                        "progress:\t{wide_bar:.green} [{pos:>7}/{len:>7}]",
+                    )?)
+                    .with_position(repair_progress.current_item as u64);
+
+                    bar.abandon();
+                }
+            }
+        }
+
+        println!();
+    }
+
+    Ok(())
+}
+
+/// Show details for a single region replacement
+async fn cmd_db_region_replacement_info(
+    _opctx: &OpContext,
+    datastore: &DataStore,
+    _fetch_opts: &DbFetchOptions,
+    args: &RegionReplacementInfoArgs,
+) -> Result<(), anyhow::Error> {
+    let request: Option<RegionReplacement> = datastore
+        .pool_connection_for_tests()
+        .await?
+        .transaction_async(|conn| async move {
+            use db::schema::region_replacement::dsl;
+
+            dsl::region_replacement
+                .filter(dsl::id.eq(args.replacement_id))
+                .limit(i64::from(1))
+                .select(RegionReplacement::as_select())
+                .first_async(&conn)
+                .await
+                .optional()
+        })
+        .await?;
+
+    let Some(request) = request else {
+        bail!(
+            "no region replacement request with id {} found",
+            args.replacement_id,
+        );
+    };
+
+    // Show details
+    println!("      started: {}", request.request_time);
+    println!("        state: {:?}", request.replacement_state);
+    println!("old region id: {}", request.old_region_id);
+    println!("new region id: {:?}", request.new_region_id);
+    println!();
+
+    if let Some(new_region_id) = request.new_region_id {
+        // Find all related notifications
+        let notifications: Vec<UpstairsRepairNotification> = datastore
+            .pool_connection_for_tests()
+            .await?
+            .transaction_async(|conn| async move {
+                use db::schema::upstairs_repair_notification::dsl;
+
+                dsl::upstairs_repair_notification
+                    .filter(dsl::region_id.eq(new_region_id))
+                    .order_by(dsl::time.asc())
+                    .select(UpstairsRepairNotification::as_select())
+                    .get_results_async(&conn)
+                    .await
+            })
+            .await?;
+
+        #[derive(Tabled)]
+        #[tabled(rename_all = "SCREAMING_SNAKE_CASE")]
+        struct Row {
+            pub time: DateTime<Utc>,
+
+            pub repair_id: String,
+            pub repair_type: String,
+
+            pub upstairs_id: String,
+            pub session_id: String,
+
+            pub notification_type: String,
+        }
+
+        let mut rows = Vec::with_capacity(notifications.len());
+
+        for notification in &notifications {
+            rows.push(Row {
+                time: notification.time,
+                repair_id: notification.repair_id.to_string(),
+                repair_type: format!("{:?}", notification.repair_type),
+                upstairs_id: notification.upstairs_id.to_string(),
```

Comment:
What is `upstairs_id` here?

Reply:
It's from whatever Upstairs is sending the notification.